internal/core/validation.go

```diff
@@ -67,7 +68,18 @@ func MustNotHaveSurroundingSpaces(val string) error {
 
 // MustBePosixAccountName is a h.ValidationRule.
 func MustBePosixAccountName(val string) error {
-	if posixAccountNameRx.MatchString(val) {
+    // fetch possible username regex from the environment
+    posixAccRxEnv := os.Getenv("PORTUNUS_POSIX_ACCOUNT_REGEX") 
+
+    // default regex is the minimally allowed unix username
+    var posix_regex *regexp.Regexp = posixAccountNameRx; 
+    
+    if posixAccRxEnv != "" {
+        // takes the fetched REGEX from the environment and builds it
+        posix_regex = regexp.MustCompile(`^` + posixAccRxEnv + `$`);
+    }
```

Comment:
Calling `MustCompile` during a request handler is a smell. That means that invalid configuration will seemingly work as long as nobody creates or updates any users, and then it blows away the entire process when someone does try to do so.

Since the engine object is the only plausible place to put it, I recommend:

- reading this env var in the Engine constructor (`core.RunEngineAsync`)
- adding a method to the Engine interface that satisfies the ValidationRule interface, e.g. `CheckUserNameSyntax(val string) error` -- then `e.CheckUserNameSyntax` can be put in the list of validation rules for the login name field

Reply:
The second big issue here is that invalid LDAP direcotry contents can be generated if the username regex is altered from its default, a non-POSIX username is used, but then the "Is POSIX user" attribute is also set.

There should be an additional cross-field validation (i.e. in `validateCreateUserForm` and `validateEditUserForm`) to ensure that users with the "Is POSIX user" attribute have a POSIX-conformant username.